File: c#/383-Ransom-Note.cs
public class Solution
{
    public bool CanConstruct(string ransomNote, string magazine)
    {
        int[] counter = new int[26];

        foreach (char c in magazine)
            counter[c - 'a']++;
        foreach (char c in ransomNote)
            if (counter[c - 'a']-- == 0)
                return false;
        return true;
    }
}

File: python/383-Ransom-Note.py
from collections import defaultdict


class Solution:

    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        magCount = defaultdict(lambda: 0)

        for char in magazine:
            magCount[char] += 1

        for char in ransomNote:
            magCount[char] -= 1
            if magCount[char] < 0:
                return False
        return True
